TeachingRecordSystem/src/TeachingRecordSystem.Core/Events/EventBase.cs

```diff
@@ -49,4 +49,22 @@ public static EventBase Deserialize(string payload, string eventName)
 
         return (EventBase)JsonSerializer.Deserialize(payload, eventType, JsonSerializerOptions)!;
     }
+
+    public bool TryGetPersonId(out Guid personId)
+    {
+        if (this is IEventWithPersonId { PersonId: var eventPersonId })
+        {
+            personId = eventPersonId;
+            return true;
+        }
+
+        if (this is IEventWithOptionalPersonId { PersonId: { } eventOptionalPersonId })
+        {
+            personId = eventOptionalPersonId;
+            return true;
+        }
+
+        personId = Guid.Empty;
```

Comment:
Shouldn't this return a nullable Guid?

Reply:
The caller could then convert that to an empty Guid if needed

Reply:
It's more jarring to use if it's nullable; this is similar to `int.TryParse()` which don't output a nullable, even when parsing fails.